jerry-core/include/jerryscript-compiler.h

```diff
@@ -47,6 +47,11 @@ extern "C"
 
 #endif /* __GNUC__ */
 
+#ifdef _MSC_VER
+void * __cdecl _alloca (size_t _Size);
+#define JERRY_ALLOCA(name, type, size) type *name = (type *)(_alloca (sizeof(type)*size))
```

Comment:
Note 1: Cppcheck apparently doesn't check the contents of the macro but please follow JerryScript coding guidelines there, too: `type *name = (type *) (_alloca (sizeof (type) * size))` (If I'm not mistaken. Reference: https://github.com/jerryscript-project/jerryscript/blob/master/docs/08.CODING-STANDARDS.md )

Note 2: In a previous version of this PR, weren't there definitions for some attributes, too? According to the docs, MSVC has declspecs like `align`, `deprecated`, `noinline`, `noreturn`. Can they be applied here?

Note 3: In the other PR, it's been discussed whether ALLOCA or VLA is the name to use. I'm not sure whether we've reached conclusion but let's keep in mind that the nema still might change. (@zherczeg, opinion?)

Note 4: Current param order of the macro is `(name, type, size)`. In both implementations (MSVC, GCC) the order of syntactic elements is `type ... name ... size` so it would be more natural for the macro arguments to follow this order, too.

Reply:
For the record: I also vote for VLA